src/components/mods/RandomHeldItemsMod.ts

```diff
@@ -0,0 +1,180 @@
+import { Component } from "eightbittr/lib/Component";
+import { ICallbackRegister, IMod } from "modattachr/lib/IModAttachr";
+
+import { FullScreenPokemon } from "../../FullScreenPokemon";
+import { INewPokemon } from "../constants/Pokemon";
+
+interface IItemProbabilities {
+    /**
+     * Name of item.
+     */
+    name: string;
+
+    /**
+     * Probability of Pokemon having item.
+     */
+    probability: number;
+}
+
+ /**
+  * Mod that randomizes items found on wild Pokemon.
+  */
+export class RandomHeldItemsMod<TGameStartr extends FullScreenPokemon> extends Component<TGameStartr> implements IMod {
+     /**
+      * Name of the mod.
+      */
+     public readonly name: string = "Random Held Items";
+
+     /**
+      * Mod events, keyed by name.
+      */
+     public readonly events: ICallbackRegister = {
+         onRandomItems: (chosenInfo: INewPokemon): INewPokemon => {
+             const pokemonName: string = chosenInfo.title.join("");
+             const pokemonType: string = this.gameStarter.constants.pokemon.byName[pokemonName].types[0];
+             const generatedNumber = this.gameStarter.numberMaker.randomReal1();
+             let counter: number = 0;
+             for (const chosenObject of this.typeItems[pokemonType]) {
```

Comment:
Move the logic in and around this for loop into a private function. That'll make it easier to read through the mod's logic.